desktop/cmp/panel/PanelModel.ts

```diff
@@ -227,8 +235,8 @@ export class PanelModel extends HoistModel {
         this.refreshMode = refreshMode;
         this.showSplitter = showSplitter;
         this.showSplitterCollapseButton = showSplitterCollapseButton;
-        this.showHeaderCollapseButton = showHeaderCollapseButton;
-        this.showModalToggleButton = showModalToggleButton;
+        this.showHeaderCollapseButton = collapsible && showHeaderCollapseButton;
+        this.showModalToggleButton = modalSupport && showModalToggleButton;
```

Comment:
I think it's an improvement to force these to false when never gonna be supported - cuts down on checking code elsewhere.  Unfortunately there's some asymmetry in that showModalToggle will always be correct, but showHeaderCollapseButton must still be combined with `!isModal` within `PanelHeader`.

Could have saved the supplied config to a _showHeaderCollapse and added a getter, wasn't sure it was worth it.